packages/itwin/grouping-mapping-widget/src/widget/components/PropertyMenu.tsx

```diff
@@ -2,404 +2,212 @@
+    groupProperties,
+    setGroupProperties,
+    calculatedProperties,
+    setCalculatedProperties,
+    customCalculationProperties,
+    setCustomCalculationProperties,
+  } = usePropertiesContext();
   const [isInformationPanelOpen, setIsInformationPanelOpen] =
     useState<boolean>(false);
-  const [resolvedHiliteIds, setResolvedHiliteIds] = useState<string[]>([]);
-  const [keySet, setKeySet] = useState<KeySet>();
-  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
 
   const fetchGroupProperties = useMemo(
     () => {
       return async () => mappingClient.getGroupProperties((await getAccessToken()), iModelId, mappingId, groupId);
     },
     [getAccessToken, mappingClient, iModelId, mappingId, groupId],
   );
-  const { isLoading: isLoadingGroupProperties, data: groupProperties, refreshData: refreshGroupProperties } =
-    useCombinedFetchRefresh<IGroupPropertyTyped>(fetchGroupProperties);
+  const { isLoading: isLoadingGroupProperties, refreshData: refreshGroupProperties } =
+    useCombinedFetchRefresh<GroupProperty>(fetchGroupProperties, setGroupProperties);
 
   const fetchCalculatedProperties = useMemo(
     () => {
       return async () => mappingClient.getCalculatedProperties((await getAccessToken()), iModelId, mappingId, groupId);
     },
     [getAccessToken, mappingClient, iModelId, mappingId, groupId],
   );
-  const { isLoading: isLoadingCalculatedProperties, data: calculatedProperties, refreshData: refreshCalculatedProperties } =
-    useCombinedFetchRefresh<ICalculatedPropertyTyped>(fetchCalculatedProperties);
+  const { isLoading: isLoadingCalculatedProperties, refreshData: refreshCalculatedProperties } =
+    useCombinedFetchRefresh<CalculatedProperty>(fetchCalculatedProperties, setCalculatedProperties);
 
   const fetchCustomCalculations = useMemo(
     () => {
       return async () => mappingClient.getCustomCalculations((await getAccessToken()), iModelId, mappingId, groupId);
     },
     [getAccessToken, mappingClient, iModelId, mappingId, groupId],
   );
-  const { isLoading: isLoadingCustomCalculations, data: customCalculations, refreshData: refreshCustomCalculations } =
-    useCombinedFetchRefresh<ICustomCalculationTyped>(fetchCustomCalculations);
+  const { isLoading: isLoadingCustomCalculations, refreshData: refreshCustomCalculations } =
+    useCombinedFetchRefresh<CustomCalculation>(fetchCustomCalculations, setCustomCalculationProperties);
 
   useEffect(() => {
     const initialize = async () => {
-      try {
-        const query = group.query;
-        const queryRowCount = await iModelConnection.queryRowCount(query);
-        if (queryRowCount === 0) {
-          toaster.warning("The query is valid but produced no results.");
-          await goBack();
-        }
-        const keys = await manufactureKeys(query, iModelConnection);
-        setKeySet(keys);
+      if (!iModelConnection) return;
+      setIsLoading(true);
```

Comment:
Will the user be able to modify properties while the elements within a group are being loaded? Not all properties require visualizing all group elements.

Reply:
Yes.